melange-core/c-lexer-cpp.dylan

```diff
@@ -130,7 +131,11 @@ define function get-macro-params
       <rparen-token> => paren-count := paren-count - 1;
       otherwise => #f;
     end select;
+    empty-token := #f
   finally
+    if (empty-token) // Insert an empty token.
+      list := pair(make(<identifier-token>, string: "", generator: state), list)
+    end if;
```

Comment:
The for loop above (starting at line 123) recursively adds tokens at the head of the list, until it determines the end of the list has been reached.

If no token was inserted in that loop, I add the empty token at the head of the list, where the argument would have been.